feature-libs/checkout/b2b/components/checkout-payment-type/checkout-payment-type.component.ts

```diff
@@ -42,22 +42,51 @@ export class CheckoutPaymentTypeComponent {
   paymentTypes$: Observable<PaymentType[]> =
     this.checkoutPaymentTypeFacade.getPaymentTypes();
 
-  typeSelected$: Observable<PaymentType> = this.checkoutPaymentTypeFacade
-    .getSelectedPaymentTypeState()
-    .pipe(
+  typeSelected$: Observable<PaymentType> = combineLatest([
+    this.checkoutPaymentTypeFacade.getSelectedPaymentTypeState().pipe(
       filter((state) => !state.loading),
-      map((state) => state.data),
-      filter(isNotUndefined),
-      distinctUntilChanged(),
-      tap((selected) => {
-        this.typeSelected = selected?.code;
-        this.checkoutStepService.resetSteps();
-        this.checkoutStepService.disableEnableStep(
-          CheckoutStepType.PAYMENT_DETAILS,
-          selected?.code === B2BPaymentTypeEnum.ACCOUNT_PAYMENT
-        );
-      })
-    );
+      map((state) => state.data)
+    ),
+    this.paymentTypes$,
+  ]).pipe(
+    map(([prevSelected, availablePaymentTypes]) => {
+      if (prevSelected) {
+        if (
+          availablePaymentTypes.find((availablePaymentType) => {
+            return availablePaymentType.code === prevSelected.code;
+          })
+        ) {
+          return prevSelected;
+        }
+      }
+      if (availablePaymentTypes.length !== 0) {
+        this.busy$.next(true);
+        this.checkoutPaymentTypeFacade
```

Comment:
We are re-using a similar part of code in the file on lines 133:141.
What do you think to delegate this into a separate method? And call this method instead in both places but with different arguments.

Reply:
Line 133:141 having below which goes to different route
` complete: () => this.checkoutStepService.next(this.activatedRoute),
        error: () => this.onError(),`
but Line 67:75
` complete: () => this.onSuccess(),
              error: () => this.onError(),`
this makes both call different.